src/firefly/js/visualize/iv/HiPSTileDrawer.js

```diff
@@ -0,0 +1,268 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+import {isNil, isString} from 'lodash';
+import {createImageUrl,initOffScreenCanvas, computeBounding, isQuadTileOnScreen, drawEmptyRecTile} from './TileDrawHelper.jsx';
+import {primePlot} from '../PlotViewUtil.js';
+import {retrieveAndProcessImage} from './ImageProcessor.js';
+import {drawOneHiPSTile} from './HiPSTileDraw.js';
+import {getVisibleHiPSCells, getPointMaxSide, getBestHiPSlevel, makeHiPSAllSkyUrl} from '../HiPSUtil.js';
+import {loadImage} from '../../util/WebUtil.js';
+import {CysConverter} from '../CsysConverter.js';
+import {findAllSkyImage, findTileImage, addAllSkyImage, addTileImage} from './HiPSTileCache.js';
+
+
+const noOp= { drawerTile : () => undefined, abort : () => undefined };
+
+
+
+
+/**
+ * Return a function that should be called on every render to draw the image
+ * @param targetCanvas
+ * @return {*}
+ */
+export function initHiPSDrawer(targetCanvas) {
+    if (!targetCanvas) return () => undefined;
+    // let cachedImages= [];
+    // let cachedAllSkyImage= {url:null, image: null};
+    let abortLastDraw= null;
+
+
+    return (plot, opacity,plotView, tileProcessInfo= {shouldProcess:false}) => {
+        if (abortLastDraw) abortLastDraw();
+        const scale = plot.zoomFactor;
+
+        const rootPlot= primePlot(plotView); // bounding box should us main plot not overlay plot
+
+        const {viewDim}= plotView;
+        const boundingBox= computeBounding(rootPlot,viewDim.width,viewDim.height);
+        const offsetX= boundingBox.x>0 ? boundingBox.x : 0;
+        const offsetY= boundingBox.y>0 ? boundingBox.y : 0;
+
+
+        let tryHipsAllsky= false;
+        let nside= getBestHiPSlevel(plot, true);
+        if (nside==='allsky') {
+            tryHipsAllsky= true;
+            nside= 3;
+        }
+
+        const cc= CysConverter.make(plot);
+
+        const {fov,centerWp}= getPointMaxSide(plot,viewDim);
+
+        const cells= getVisibleHiPSCells(nside,centerWp, fov);
+
+        const tilesToLoad= cells
+            .map( (c) => {
+                const devPtCorners= c.wpCorners.map( (corner) => cc.getDeviceCoords(corner));
+                if (!isQuadTileOnScreen(devPtCorners, viewDim)) return null;
+                return {devPtCorners, tileNumber:c.ipix, dx:0, dy:0, nside};
+            })
+            .filter( (c) => c);
+
+        const drawer= makeHipsDrawer(plotView, plot, targetCanvas, tilesToLoad.length,
+            offsetX,offsetY, scale, opacity, tileProcessInfo);
+
+        if (tryHipsAllsky) {
+            const allSkyURL= makeHiPSAllSkyUrl(plot);
+            const cachedAllSkyImage= findAllSkyImage(allSkyURL);
+            if (cachedAllSkyImage) {
+                drawAllSky(cachedAllSkyImage, tilesToLoad, drawer);
+            }
+            else {
+                loadImage(makeHiPSAllSkyUrl(plot))
```

Comment:
Can you use allSkyURL constant here?

Reply:
The allsky image differs depending on the dataset. It is easier to remake the string that having to watch when the name changes.